engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodNode.java

```diff
@@ -169,16 +168,18 @@ Function resolveFunction(
       return null;
     }
 
-    RootNode where = function.getCallTarget().getRootNode();
-    // If both Any and the type where `function` is declared, define `symbol`
-    // and the method is invoked statically, i.e. type of self is the eigentype,
-    // then we want to disambiguate method resolution by always resolved to the one in Any.
-    EnsoContext ctx = EnsoContext.get(this);
-    if (where instanceof MethodRootNode node && typeCanOverride(node, ctx)) {
-      Type any = ctx.getBuiltins().any();
-      Function anyFun = symbol.getScope().lookupMethodDefinition(any, symbol.getName());
-      if (anyFun != null) {
-        function = anyFun;
+    if (selfTpe.getDefinitionScope().getAssociatedType() != selfTpe) {
```

Comment:
Refines
- #6441
- #7033 

to give precedence to `Meta.is_a` over `Any.is_a self`, just like it has been done when `Meta.is_a` was a builtin.

Reply:
- separated into its own #12048